Makefile

```diff
@@ -0,0 +1,378 @@
+
+# https://github.com/jetstack/cert-manager/releases
+# ⚠️  You may want to keep this in sync with RABBITMQ_TOPOLOGY_OPERATOR_VERSION
+# In other words, don't upgrade cert-manager to a version that was released AFTER RABBITMQ_TOPOLOGY_OPERATOR_VERSION
+CERT_MANAGER_VERSION ?= 1.5.4
+install-cert-manager: | $(KUBECONFIG) $(KUBECTL) ## Install Cert Manager - dependency of RabbitMQ Topology Operator
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/jetstack/cert-manager/releases/download/v$(CERT_MANAGER_VERSION)/cert-manager.yaml
+	$(KUBECTL) wait --for=condition=available deploy/cert-manager-webhook --timeout=60s --namespace $(CERT_MANAGER_NAMESPACE)
+
+# https://github.com/rabbitmq/messaging-topology-operator/releases
+RABBITMQ_TOPOLOGY_OPERATOR_VERSION ?= 1.2.1
+install-rabbitmq-topology-operator: | install-cert-manager $(KUBECTL) ## Install RabbitMQ Topology Operator
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/rabbitmq/messaging-topology-operator/releases/download/v$(RABBITMQ_TOPOLOGY_OPERATOR_VERSION)/messaging-topology-operator-with-certmanager.yaml
+
+KNATIVE_VERSION ?= 1.0.0
+
+# https://github.com/knative/serving/releases
+install-knative-serving: | $(KUBECONFIG) $(KUBECTL) ## Install Knative Serving
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/serving/releases/download/knative-v$(KNATIVE_VERSION)/serving-crds.yaml
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/serving/releases/download/knative-v$(KNATIVE_VERSION)/serving-core.yaml
+	$(KUBECTL) wait --for=condition=available deploy/controller --timeout=60s --namespace $(SERVING_NAMESPACE)
+	$(KUBECTL) wait --for=condition=available deploy/webhook --timeout=60s --namespace $(SERVING_NAMESPACE)
+	$(KUBECTL) apply --filename https://github.com/knative/net-kourier/releases/download/knative-v$(KNATIVE_VERSION)/kourier.yaml
+	$(KUBECTL) patch configmap/config-network --namespace $(SERVING_NAMESPACE) --type merge \
+		--patch '{"data":{"ingress.class":"kourier.ingress.networking.knative.dev"}}'
+
+# https://github.com/knative/eventing/releases
+install-knative-eventing: | $(KUBECONFIG) $(KUBECTL) ## Install Knative Eventing
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/eventing/releases/download/knative-v$(KNATIVE_VERSION)/eventing-crds.yaml
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/eventing/releases/download/knative-v$(KNATIVE_VERSION)/eventing-core.yaml
+	$(KUBECTL) wait --for=condition=available deploy/eventing-controller --timeout=60s --namespace $(EVENTING_NAMESPACE)
+	$(KUBECTL) wait --for=condition=available deploy/eventing-webhook --timeout=60s --namespace $(EVENTING_NAMESPACE)
+
+install: | $(KUBECONFIG) $(KO) install-knative-serving install-knative-eventing install-rabbitmq-cluster-operator install-rabbitmq-topology-operator ## Install local dev Knative Eventing RabbitMQ - manages all dependencies, including K8s components
```

Comment:
This script is sooo long. Why will be maintaining this? There are 20-30 knative repos. Are we realy like to copy and paste this so many times, and change couple of things, that needs adjusting, in between?!?

Reply:
I understand that your perspective is the wider Knative ecosystem and I must agree that I have not thought about the big picture since my focus was to get this working in the context of this repository. I recognise the tension between _Make it work_ vs _Make it right_ and while I welcome the second challenge, the focus is on small scope and trying to understand if this approach will even work.

I must emphasise that this is not a script, it's a collection of commands chained together. The distinction may seem insignificant, but it's an important mind shift for approaching Make the right way. This example illustrates Make approached from a different perspective which is valid for that context, but would be the wrong approach for Knative in my opinion:

https://github.com/ninenines/erlang.mk/blob/d80984c1036ea81eb1f44b8d7cde85fc09b5e3c0/core/deps.mk#L197-L248